### PR TITLE
YOLO Model Training Pipeline

### PR DESCRIPTION
**Pull request by ArthurGuihaire**
## Main changes:
- Create python scripts for training YOLO models locally, for real or synthetic data:
    - organize_dataset.py splits the data into train/test/val
    - training.py trains a YOLO model on real data using a specified base model
    - training.sh runs the whole training pipeline for fine-tuning YOLO model on real data based on the synthetic model
- Create scripts for augmenting images
    - augment_data.py aggressively augments data, used to train on simulation data for fine-tuning later. This should only be ran on train sets, not val or test
- Document the two training pipelines
- Document how to fine-tune unity models on real data

## Other random scripts:
- fix_labels.py remaps label files from unity labels to real data labels, since the format is different. It also updates the corresponding data.yaml
- visualize_label.py can be used to visualize a single label file, drawing the bounding boxes on top of the image

### CODE REVIEW

**Review by avyan-k — DISMISSED** [review]
Looks great, I tested the fine-tuning with the dataset on the drive and it went pretty smoothly (there are some folder organization issues with the zip file on the drive but that has nothing to do with this PR). The steps are pretty well outlined in the documentation so I do not expect confusion at comp.

File: ros2_ws/src/vision/model_pipeline/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+4. **Train Model**: Run the training script on the processed dataset.
+   ```bash
+   python3 training.py --model v11 --size s --data data/processed_aug/data.yaml --unity
+   ```
+
+## Fine-tuning Pipeline (Local)
+1. Collect real data
+    1. Ideally, you should collect at least ~50 raw images for each class. 
+2. Upload and Label Data
+    1. Use [this slideshow](https://docs.google.com/presentation/d/12WbaJi48y1edtK-ood4Iyu_NI00ZPQlZVDVObNEsrkc/edit?slide=id.p#slide=id.p) to sign in to roboflow
```
Comment (WilliamZhang2205) [inline ros2_ws/src/vision/model_pipeline/README.md]: We can remove the login credentials for roboflow and the email account. Instead, add a path from the robotics google drive?

File: ros2_ws/src/vision/model_pipeline/README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    2. Follow the guide in the slideshow on how to label images
+3. Generate a New Version of the Dataset
+    1. Do **NOT** add any pre-processing/augmentation; this is handled in the training notebook.
+    2. The name of the dataset version does not matter.
+4. On the new dataset version, click download dataset
```
Comment (WilliamZhang2205) [inline ros2_ws/src/vision/model_pipeline/README.md]: Maybe add the export format (I assumed yolov11 for this review) and "to download as zip to computer" in the Roboflow pop up that comes up when downloading dataset.

File: ros2_ws/src/vision/model_pipeline/README.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+| `-e` | `--epochs` | Number of training epochs | `200` |
+| `-b` | `--batch` | Batch size. -1 for auto | `-1` |
+| `-lr0` | `--learning-rate` | Initial learning rate | `0.0003` |
+|  | `--imgsz` | Image size for training | `640` |
+|  | `--workers` | Number of dataloader workers | `2` |
+|  | `--cache` | Cache images for faster training | `False` |
+
+To pass any of these arguments, simply add them to the end of the `training.sh` command. For example, if you wanted to train a medium model with 90% training and 5% validation for 1 epoch, you would run the following command: `
+```bash
+./training.sh yolov11s_augmented_synthetic_best_model.pt "--size m --train 0.9 --val 0.05" --training-args "--epochs 1"
```
Comment (WilliamZhang2205) [inline ros2_ws/src/vision/model_pipeline/README.md]: add the --organize-args to this example command

File: ros2_ws/src/vision/model_pipeline/augment_dataset.py [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+            # print(f"Error augmenting {image_path.name}: {e}")
+            continue
+            
+    return count
+
+def run_visualization(img_files, lbl_input_dir, output_dir, transform, num_vis):
+    vis_dir = output_dir / "visualization"
+    vis_dir.mkdir(parents=True, exist_ok=True)
+    
+    import random
```
Comment (WilliamZhang2205) [inline ros2_ws/src/vision/model_pipeline/augment_dataset.py]: why is there an import here? Isnt it also already at the top

**Comment by kaidalisohaib** [comment]
yo
